File: Samples/azure-storage/Azure.Java.Fluent/implementation/StorageAccountUpdateParametersInner.java
/**
 */

package petstore.implementation;

import java.util.Map;
import petstore.StorageAccountPropertiesUpdateParameters;
import com.fasterxml.jackson.annotation.JsonProperty;

/**
 * The parameters to update on the account.
 */
public class StorageAccountUpdateParametersInner {
    /**
     * Resource tags.
     */
    @JsonProperty(value = "tags")
    private Map<String, String> tags;

    /**
     * The properties property.
     */
    @JsonProperty(value = "properties")
    private StorageAccountPropertiesUpdateParameters properties;

    /**
     * Get the tags value.
     *
     * @return the tags value
     */
    public Map<String, String> tags() {
        return this.tags;
    }

    /**
     * Set the tags value.
     *
     * @param tags the tags value to set
     * @return the StorageAccountUpdateParametersInner object itself.
     */
    public StorageAccountUpdateParametersInner withTags(Map<String, String> tags) {
        this.tags = tags;
        return this;
    }

    /**
     * Get the properties value.
     *
     * @return the properties value
     */
    public StorageAccountPropertiesUpdateParameters properties() {
        return this.properties;
    }

    /**
     * Set the properties value.
     *
     * @param properties the properties value to set
     * @return the StorageAccountUpdateParametersInner object itself.
     */
    public StorageAccountUpdateParametersInner withProperties(StorageAccountPropertiesUpdateParameters properties) {
        this.properties = properties;
        return this;
    }

}

File: Samples/azure-storage/Azure.CSharp.Fluent/Models/StorageAccountUpdateParametersInner.cs

namespace Petstore.Models
{
    using Newtonsoft.Json;
    using System.Collections;
    using System.Collections.Generic;
    using System.Linq;

    /// <summary>
    /// The parameters to update on the account.
    /// </summary>
    public partial class StorageAccountUpdateParametersInner
    {
        /// <summary>
        /// Initializes a new instance of the
        /// StorageAccountUpdateParametersInner class.
        /// </summary>
        public StorageAccountUpdateParametersInner()
        {
          CustomInit();
        }

        /// <summary>
        /// Initializes a new instance of the
        /// StorageAccountUpdateParametersInner class.
        /// </summary>
        /// <param name="tags">Resource tags</param>
        public StorageAccountUpdateParametersInner(IDictionary<string, string> tags = default(IDictionary<string, string>), StorageAccountPropertiesUpdateParameters properties = default(StorageAccountPropertiesUpdateParameters))
        {
            Tags = tags;
            Properties = properties;
            CustomInit();
        }

        /// <summary>
        /// An initialization method that performs custom operations like setting defaults
        /// </summary>
        partial void CustomInit();

        /// <summary>
        /// Gets or sets resource tags
        /// </summary>
        [JsonProperty(PropertyName = "tags")]
        public IDictionary<string, string> Tags { get; set; }

        /// <summary>
        /// </summary>
        [JsonProperty(PropertyName = "properties")]
        public StorageAccountPropertiesUpdateParameters Properties { get; set; }

        /// <summary>
        /// Validate the object.
        /// </summary>
        /// <exception cref="Microsoft.Rest.ValidationException">
        /// Thrown if validation fails
        /// </exception>
        public virtual void Validate()
        {
            if (Properties != null)
            {
                Properties.Validate();
            }
        }
    }
}
